cmd/query/app/querysvc/query_service.go

```diff
@@ -134,6 +136,7 @@ func (qs QueryService) GetDependencies(ctx context.Context, endTs time.Time, loo
 func (qs QueryService) GetCapabilities() StorageCapabilities {
 	return StorageCapabilities{
 		ArchiveStorage: qs.options.hasArchiveStorage(),
+		MetricStorage:  qs.options.hasMetricStorage(),
```

Comment:
why does it need to interrogate Options if you have access to `qs` which likely already stores the reference to metric reader?

Reply:
No qs doesn't store the reference to the metrics reader. The options does.